src/app/public/modules/colorpicker/colorpicker.component.spec.ts

```diff
@@ -290,17 +294,52 @@ describe('Colorpicker Component', () => {
       });
     }));
 
-    it('should add aria-label to input if not specified', fakeAsync(() => {
+    it('should add aria-label and title attributes to button if not specified', fakeAsync(() => {
+      const defaultLabel = 'Select color value';
+      fixture.detectChanges();
+      tick();
+
+      expect(getColorpickerButton(nativeElement).getAttribute('aria-label')).toBe(defaultLabel);
+      expect(getColorpickerButton(nativeElement).getAttribute('title')).toBe(defaultLabel);
+    }));
+
+    it('should allow consumer to override aria-label and title attributes', fakeAsync(() => {
+      const newLabel = 'FOOBAR';
+      component.label = newLabel;
+
       fixture.detectChanges();
       tick();
-      expect(nativeElement.querySelector('input').getAttribute('aria-label')).toBe('Color value');
+
+      expect(getColorpickerButton(nativeElement).getAttribute('aria-label')).toBe(newLabel);
+      expect(getColorpickerButton(nativeElement).getAttribute('title')).toBe(newLabel);
     }));
 
-    it('should not add aria-label to input when one was specified', fakeAsync(() => {
-      nativeElement.querySelector('input').setAttribute('aria-label', 'Best picker');
+    it('should not provide aria-labelledby attribute to button if none is specified', fakeAsync(() => {
       fixture.detectChanges();
       tick();
-      expect(nativeElement.querySelector('input').getAttribute('aria-label')).toBe('Best picker');
+
+      expect(getColorpickerButton(nativeElement).getAttribute('aria-labelledby')).toBeNull();
+    }));
+
+    it('should add aria-labelledby attribute to button if provided by consumer', fakeAsync(() => {
+      const customLabelledBy = 'myId';
+      component.labelledBy = customLabelledBy;
+
+      fixture.detectChanges();
+      tick();
+
+      expect(getColorpickerButton(nativeElement)
+        .getAttribute('aria-labelledby')).toBe(customLabelledBy);
+    }));
+
+    it('should not set label and title attributes if labelledby is provided by consumer', fakeAsync(() => {
```

Comment:
Should we have a test which test if both `label` and `labelledBy` is given? I know that maybe this doesn't make a ton of sense; however, looking at the code it is technically possible that the label is used for a `title` and then `labelledBy` is also given.

Reply:
That actually brings up a good point - is there ever a time where this would make sense and is that going to be problemnatic that we are using the same property for both `aria-label` and `title`? The only way to set the `title` is to use `label` and if you also give a `labelledBy` then both `aria-label` and `aria-labelledBy` will be set.

Reply:
I see we call out not doing this in the JS Docs - just wanted to ask the question about if there is ever a case where this is appropriate.

Reply:
There's nothing stopping a user from providing both `label` and `labelledby`. While it's not preferred, I don't think we prevent it in any of our other components. What would you be wanting to test here?

Reply:
Basically that if you have both that the `label` takes precedence in your `||` statements in the template. 

I'm not sure that having both would pass `aXe` testing - but it is technically possible.